.editorconfig

```diff
@@ -0,0 +1,26 @@
+# EditorConfig helps developers define and maintain consistent
+# coding styles between different editors and IDEs
+# editorconfig.org
+
+root = true
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+
+[*.rs]
+max_line_length = 100
```

Comment:
this is taken from rust's own editorconfig, unsure if this is what we want: https://github.com/rust-lang/rust/blob/master/.editorconfig